src/gleam_stdlib.erl

```diff
@@ -411,6 +426,38 @@ inspect(Any) when is_function(Any) ->
 inspect(Any) ->
     ["//erl(", io_lib:format("~p", [Any]), ")"].
 
+inspect_maybe_gleam_atom([], none, []) ->
+    {error, cannot_be_an_empty_string};
+inspect_maybe_gleam_atom([Head | _Rest], none, []) when ?is_digit_char(Head) ->
+    {error, cannot_start_with_a_digit};
+inspect_maybe_gleam_atom([$_ | _Rest], none, []) ->
+    {error, cannot_start_with_an_underscore};
+inspect_maybe_gleam_atom([$_ | []], _PrevChar, _Acc) ->
+    {error, cannot_end_with_an_underscore};
+inspect_maybe_gleam_atom([$_ | _Rest], $_, _Acc) ->
+    {error, cannot_contain_consecutive_underscores};
+inspect_maybe_gleam_atom([Head | _Rest], _PrevChar, _Acc)
+    when ?is_lowercase_char(Head) == false andalso ?is_underscore_char(Head) == false andalso ?is_digit_char(Head) == false ->
+    {error, need_to_only_contain_lower_case_letters_aor_digits_aor_underscores};
+% Handle first char -> uppercase and prepend
+inspect_maybe_gleam_atom([Head | Rest], none, Acc) ->
+    inspect_maybe_gleam_atom(Rest, Head, [string:uppercase([Head]) | Acc]);
+% Handle underscore -> only set as PrevChar
+inspect_maybe_gleam_atom([$_ | Rest], _PrevChar, Acc) ->
+    inspect_maybe_gleam_atom(Rest, $_, Acc);
+% Handle char after underscore -> uppercase and prepend
+inspect_maybe_gleam_atom([Head | Rest], $_, Acc) ->
+    inspect_maybe_gleam_atom(Rest, Head, [string:uppercase([Head]) | Acc]);
+% Handle char after number -> uppercase and prepend
+inspect_maybe_gleam_atom([Head | Rest], PrevChar, Acc) when ?is_digit_char(PrevChar) ->
+    inspect_maybe_gleam_atom(Rest, Head, [string:uppercase([Head]) | Acc]);
+% Handle any other char -> prepend only
+inspect_maybe_gleam_atom([Head | Rest], _PrevChar, Acc) ->
+    inspect_maybe_gleam_atom(Rest, Head, [Head | Acc]);
+% Handle end of string -> return reversed acc ok tuple
```

Comment:
Could we remove all these comments please

Reply:
I did so, I thought about adding comments for the error cases that are not not self documenting anymore, but removed them as well: https://github.com/gleam-lang/stdlib/pull/479/commits/b1c56814c8b817f30447ed107d9980321d18dcb0